session/infinispan/remote/src/test/java/org/wildfly/clustering/session/infinispan/remote/InfinispanServerContainer.java

```diff
@@ -21,38 +22,49 @@
 import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
 import org.testcontainers.utility.DockerImageName;
 
+import com.github.dockerjava.api.model.Info;
+
 /**
  * @author Paul Ferraro
  */
 public class InfinispanServerContainer extends GenericContainer<InfinispanServerContainer> implements Function<ConfigurationBuilder, RemoteCacheContainer>, Consumer<OutputFrame> {
 
 	static final DockerImageName DOCKER_IMAGE = DockerImageName.parse(System.getProperty("infinispan.server.image", "quay.io/infinispan/server:" + Version.getMajorMinor()));
+	static final Integer HOTROD_PORT = Integer.valueOf(System.getProperty("infinispan.server.port", "11222"));
+	static final String USERNAME = System.getProperty("infinispan.server.username", "admin");
+	static final String PASSWORD = System.getProperty("infinispan.server.password", "changeme");
 	static final Logger LOGGER = Logger.getLogger(InfinispanServerContainer.class);
-	static final int DEFAULT_PORT = 11222;
-	static final String DEFAULT_USER = "admin";
-	static final String DEFAULT_PASSWORD = "changeme";
+	static final Set<String> OS_TYPES_SUPPORTING_HOST_NETWORK_MODE = Set.of("linux");
+	static final String HOST_NETWORK_MODE = "host";
 
 	InfinispanServerContainer() {
 		super(DOCKER_IMAGE);
 
-		this.setNetworkMode("host");
+		Info info = this.dockerClient.infoCmd().exec();
+		// Use host networking, if possible
+		if (OS_TYPES_SUPPORTING_HOST_NETWORK_MODE.contains(info.getOsType())) {
```

Comment:
I opened https://github.com/wildfly-clustering/wildfly-clustering/pull/23 to fix this since this is useless - this is probably always linux as this is the OS of the docker host. But I see main got updated in the meantime